src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.Stream.cs

```diff
@@ -0,0 +1,347 @@
+                    // If there are unprocessed bytes, encode and write them as the final block.
+                    ReadOnlySpan<byte> bytesNotProcessedFromPreviousChunk = this.buffer.AsSpan().Slice(0, this.numBytesNotWrittenFromPreviousChunk);
+
+                    this.jsonWriter.Base64EncodeAndWriteChunk(bytesNotProcessedFromPreviousChunk, isFinalBlock: true, out numBytesNotWrittenFromPreviousChunk);
+                    this.numBytesNotWrittenFromPreviousChunk = 0;
+                }
+
+                if (this.buffer != null)
+                {
+                    ArrayPool<byte>.Shared.Return(this.buffer);
+                }
+
+                this.Flush();
+                base.Dispose(disposing);
+            }
+
+            /// <summary>
+            /// Asynchronously disposes of the current object and performs cleanup operations.
+            /// </summary>
+            /// <returns>A <see cref="ValueTask"/> representing the asynchronous operation.</returns>
+            public override async ValueTask DisposeAsync()
+            {
+                if (this.numBytesNotWrittenFromPreviousChunk > 0)
+                {
+                    // If there are unprocessed bytes, encode and write them as the final block.
+                    ReadOnlyMemory<byte> bytesNotProcessedFromPreviousChunk = this.buffer.AsMemory().Slice(0, this.numBytesNotWrittenFromPreviousChunk);
+
+                    this.jsonWriter.Base64EncodeAndWriteChunk(bytesNotProcessedFromPreviousChunk.Span, isFinalBlock: true, out numBytesNotWrittenFromPreviousChunk);
+                    this.numBytesNotWrittenFromPreviousChunk = 0;
+                }
+
+                if (this.buffer != null)
+                {
+                    ArrayPool<byte>.Shared.Return(this.buffer);
+                }
+
+                await this.jsonWriter.FlushAsync().ConfigureAwait(false);
+            }
+
+            /// <summary>
+            /// Reads a sequence of bytes from the current stream and advances the position within the stream by the numbers of bytes read. This operation is not supported by this stream.
+            /// </summary>
+            public override int Read(byte[] buffer, int offset, int count)
+            {
+                throw new NotSupportedException();
+            }
+
+            /// <summary>
+            /// Sets the position within the stream. This operation is not supported by this stream.
+            /// </summary>
+            public override long Seek(long offset, SeekOrigin origin)
+            {
+                throw new NotSupportedException();
+            }
+
+            /// <summary>
+            /// Gets the length in bytes of the stream. This is not supported by this stream.
+            /// </summary>
+            public override void SetLength(long value)
+            {
+                throw new NotSupportedException();
+            }
+
+            /// <summary>
+            /// Writes a sequence of bytes to the current stream.

```

Comment:
nit:
```suggestion
            /// Writes a portion of a byte array to the underlying stream in chunks.
```
I think it'd be better to have the sync and async method docs mirror each other